Rlabkey/R/labkey.webdav.R

```diff
@@ -0,0 +1,265 @@
+##
+#  Copyright (c) 2019 LabKey Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+##
+
+labkey.webdav.get <- function(baseUrl=NULL, folderPath, remoteFilePath, localFilePath, overwrite=TRUE, fileSet="@files")
+{
+    baseUrl=labkey.getBaseUrl(baseUrl);
+
+    ## check required parameters
+    if (missing(baseUrl) || is.null(baseUrl) || missing(folderPath) || missing(remoteFilePath) || missing(localFilePath)){
+        stop (paste("A value must be specified for each of baseUrl, folderPath, fileSet, remoteFilePath, and localFilePath"));
+    }
+
+    if (labkey.webdav.isDirectory(baseUrl = baseUrl, folderPath = folderPath, remoteFilePath = remoteFilePath, fileSet = fileSet, haltOnError = T)){
+      stop('The requested file is a directory.  Please see labkey.webdav.downloadFolder()')  
+    }
+    
+    ## normalize the folder path
+    folderPath <- encodeFolderPath(folderPath);
+    remoteFilePath <- encodeRemotePath(remoteFilePath)
+
+    url <- paste(baseUrl, "_webdav", folderPath, fileSet, "/", remoteFilePath, sep="");
+
+    response <- labkey.webdav.getByUrl(url, localFilePath, overwrite)
+
+    return(file.exists(localFilePath))
+}
+
+labkey.webdav.getByUrl <- function(url, localFilePath, overwrite=TRUE)
+{
+    # dont bother querying if this file already exists, since we wont overwrite it
+    if (!overwrite & file.exists(localFilePath)) {
+        return()
```

Comment:
Could this return an indication that it no-opped?
```suggestion
        return(FALSE)
```

Reply:
OK.  I had to change a little more to make this pass through and I also updated the man page to document that behavior.